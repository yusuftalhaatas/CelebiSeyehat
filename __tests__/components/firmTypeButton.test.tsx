import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import FirmTypeButton from '../../src/components/firmTypeButton';

describe('components/FirmTypeButton', () => {
  test('renders correctly and responds to onPress', () => {
    const onPressMock = jest.fn();
    const {getByText, toJSON} = render(
      <FirmTypeButton name="Test Button" onPress={onPressMock} />,
    );

    // snapshot test
    expect(toJSON()).toMatchSnapshot();

    // onPress event test
    fireEvent.press(getByText('Test Button'));
    expect(onPressMock).toHaveBeenCalled();
  });
});
