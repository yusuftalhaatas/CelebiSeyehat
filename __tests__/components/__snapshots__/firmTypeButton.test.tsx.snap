// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/FirmTypeButton renders correctly and responds to onPress 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "gray",
      "borderRadius": 10,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "center",
      "margin": 5,
      "opacity": 1,
      "padding": 10,
      "width": 100,
    }
  }
>
  <Text
    style={
      {
        "color": "#00ccff",
        "fontSize": 17,
        "fontWeight": "bold",
      }
    }
  >
    Test Button
  </Text>
</View>
`;
